tests/testthat/_snaps/replace.md

```diff
@@ -29,3 +29,16 @@
       Error in `str_replace_all()`:
       ! `pattern` can't be a boundary.
 
+# replacement function must return correct type/length
+
+    Code
+      str_replace_all("x", "x", ~1)
+    Condition
+      Error in `str_replace_all()`:
+      ! `replacement()` must return a character vector, not a number.
+    Code
+      str_replace_all("x", "x", ~ c("a", "b"))
+    Condition
+      Error in `str_replace_all()`:
+      ! `replacement()` must return a vector the same length as the input (1), not length 2.
```

Comment:
My brain kind of broke on this because `replacement` is really the arg name, but i see what you are trying to do by adding `()` since it is used as a function call. I don't mind either way, just wanted to say it took me a second to realize it wasn't a typo

Reply:
I'll tweak this to make it clearer.